### PR TITLE
Bug Fix - Bug fix for latest megatron-lm benchmark

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
Bug fix to sync latest megatron-lm code.

### CODE REVIEW

File: superbench/benchmarks/model_benchmarks/megatron_gpt3.py [hunk 1]
```diff
@@ -116,6 +116,7 @@ def add_parser_arguments(self):
         self._parser.add_argument('--data_home', type=str, default='/tmp', help='Data home.')
         self._parser.add_argument('--vocab_path', type=str, default='/tmp/gpt2-vocab.json', help='Vocab path.')
         self._parser.add_argument('--merge_path', type=str, default='/tmp/gpt2-merges.txt', help='Merge path.')
+        self._parser.add_argument('--split', type=str, default='949,50,1', help='Split dataset.')
```
Comment (cp5555) [inline superbench/benchmarks/model_benchmarks/megatron_gpt3.py:119]: What's the meaning of these numbers? Would you please explain it in help?

File: third_party/Makefile [hunk 2]
```diff
@@ -178,17 +178,19 @@ directx_amf_encoding_latency:
...
+	apt install -y python3-mpi4py && \
 	python -m pip install -r requirements.txt
 
 # Install Megatron-DeepSpeed
 megatron_deepspeed:
 	if [ ! -d "Megatron/Megatron-DeepSpeed" ]; then \
-        git clone "https://github.com/microsoft/Megatron-DeepSpeed.git" "Megatron/Megatron-DeepSpeed"; \
-    fi
+        	git clone -b 71e8407 "https://github.com/microsoft/Megatron-DeepSpeed.git" "Megatron/Megatron-DeepSpeed"; \
+    	fi
```
Comment (cp5555) [inline third_party/Makefile:191]: Alignment issue?

File: third_party/Makefile [hunk 3]
```diff
@@ -178,17 +178,19 @@ directx_amf_encoding_latency:
 # Install Megatron-LM
 megatron_lm:
 	if [ ! -d "Megatron/Megatron-LM" ]; then \
-        git clone "https://github.com/NVIDIA/Megatron-LM.git" "Megatron/Megatron-LM"; \
-    fi
+        	git clone -b 52b7a18 "https://github.com/NVIDIA/Megatron-LM.git" "Megatron/Megatron-LM"; \
```
Comment (cp5555) [inline third_party/Makefile:181]: Should we need to add a submodule? @abuccts

File: superbench/benchmarks/model_benchmarks/model_base.py [hunk 4]
```diff
@@ -265,8 +265,8 @@ def __train(self, precision):
         # The unit of step time should be millisecond.
         step_times = self._train_step(precision)
         if isinstance(step_times, tuple):
-            step_times = step_times[0]
             info = step_times[1]
+            step_times = step_times[0]
```
Comment (cp5555) [inline superbench/benchmarks/model_benchmarks/model_base.py:269]: Why change the order???

File: third_party/Makefile (hunk 3 above)
Comment (abuccts) [reply third_party/Makefile:181]: it's ok to add

File: superbench/benchmarks/model_benchmarks/model_base.py (hunk 4 above)
Comment (yukirora) [reply superbench/benchmarks/model_benchmarks/model_base.py:269]: there's a bug for previous order, the info will be the second step time
```
step_times = step_times[0]
info = step_times[1]
```

Comment (cp5555) [reply superbench/benchmarks/model_benchmarks/model_base.py:269]: got it

File: third_party/Makefile (hunk 3 above)
Comment (yukirora) [reply third_party/Makefile]: @abuccts, do you know how to solve this issue, when git clone submodule, it saids there's filename too long
image

Comment (abuccts) [reply third_party/Makefile]: try https://stackoverflow.com/a/22575737/7630831

**Review by cp5555 — APPROVED** [review]

**Review by abuccts — APPROVED** [review]

**Pull request merged**